lib/compiler/src/beam_ssa_type.erl

```diff
@@ -1349,6 +1361,10 @@ eval_type_test_bif(I, Op, Types) ->
             I
     end.
 
+t_is_number(#t_integer{}) -> true;
```

Comment:
I'd prefer this were added to `beam_types` where we already have similar functions. `is_numerical_type` would fit right in.